File: src/problems_all/1290_convert_binary_number_in_a_linked_list_to_integer_v1.py
"""
https://leetcode.com/problems/convert-binary-number-in-a-linked-list-to-integer/


Runtime: 33 ms
Memory: 13.9 MB
https://leetcode.com/problems/convert-binary-number-in-a-linked-list-to-integer/submissions/870483583/
"""


# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        values = []
        current = head

        while current is not None:
            values.append(str(current.val))
            current = current.next

        binary = ''.join(values)
        decimal = int(binary, 2)

        return decimal


def main():
    head = ListNode(1, next=ListNode(0, next=ListNode(1)))
    result = Solution().getDecimalValue(head)
    print(result)


if __name__ == '__main__':
    main()

File: src/problems_all/1290_convert_binary_number_in_a_linked_list_to_integer/1290_convert_binary_number_in_a_linked_list_to_integer_v1.go
/*
https://leetcode.com/problems/convert-binary-number-in-a-linked-list-to-integer/


Runtime: 0 ms
Memory: 1.9 MB
https://leetcode.com/problems/convert-binary-number-in-a-linked-list-to-integer/submissions/880728604/
*/

package main

import (
	"fmt"
	"strconv"
)

type ListNode struct {
	Val  int
	Next *ListNode
}

func main() {
	head := ListNode{1, &ListNode{0, &ListNode{1, nil}}}
	result := getDecimalValue(&head)
	fmt.Println(result)
}

func getDecimalValue(head *ListNode) int {
	binary := ""
	current := head
	for current != nil {
		binary += strconv.Itoa(current.Val)
		current = current.Next
	}

	val, err := strconv.ParseInt(binary, 2, 32)
	if err != nil {
		fmt.Println(err)
	}

	return int(val)
}
